server/src/main/java/org/opensearch/action/admin/cluster/stats/ClusterStatsIndices.java

```diff
@@ -113,21 +160,23 @@ public ClusterStatsIndices(List<ClusterStatsNodeResponse> nodeResponses, Mapping
 
                     if (shardStats.getShardRouting().primary()) {
                         indexShardStats.primaries++;
-                        docs.add(shardCommonStats.docs);
+                        docsStatsConsumer.accept(shardCommonStats.docs);
                     }
-                    store.add(shardCommonStats.store);
-                    fieldData.add(shardCommonStats.fieldData);
-                    queryCache.add(shardCommonStats.queryCache);
-                    completion.add(shardCommonStats.completion);
-                    segments.add(shardCommonStats.segments);
+                    storeStatsConsumer.accept(shardCommonStats.store);
```

Comment:
Lets add a test for the case where metrics are requested by user (with filter) but the cluster has no indices.

Reply:
Added.